src/core/Fetch.ts

```diff
@@ -0,0 +1,77 @@
+export interface RequestOptions {
+    timeout?: number;
+    data?: object;
+    headers?: Record<string, string>;
+}
+
+function queryStringify(data = {}) {
+    if (typeof data !== 'object') {
+        throw new Error('Data must be object');
+    }
+    let arr = [...Object.entries(data)];
+    return (data) ? arr.reduce((prev, item, index) => {
+        return prev += `${item[0]}=${item[1]}${(index !== arr.length - 1) ? '&' : ''}`;
+    }, '?') : '';
+}
+
+enum METHODS {
+    GET = 'GET',
+    POST = 'POST',
+    PUT = 'PUT',
+    DELETE = 'DELETE'
+}
+
+type Options = {
+    method: string;
+    timeout?: number;
+    data?: any;
+    headers?: { [key: string]: string };
+}
+
+type OptionsWithoutMethod = Omit<Options, 'method'>;
+
+export default class Fetch {
+    get(url: string | URL, options: OptionsWithoutMethod = {}): Promise<XMLHttpRequest> {
+        return this.request(url + queryStringify(options.data), {...options, method: METHODS.GET}, options.timeout);
+    }
+
+    post(url: string | URL, options: Options = {method: METHODS.POST}) {
+        return this.request(url, {...options, method: METHODS.POST}, options.timeout);
+    }
+
+    put(url: string | URL, options: Options = {method: METHODS.PUT}) {
+        return this.request(url, {...options, method: METHODS.PUT}, options.timeout);
+    }
+
+    delete(url: string | URL, options: Options = {method: METHODS.DELETE}) {
```

Comment:
Метод ниже передается, здесь излишне
```suggestion
    delete(url: string | URL, options: Options = {}) {
```